collect_app/src/main/java/org/odk/collect/android/javarosawrapper/FormController.java

```diff
@@ -183,7 +183,7 @@ public AuditEventLogger getAuditEventLogger() {
      */
     public String getXPath(FormIndex index) {
         String value;
-        switch (getEvent()) {
+        switch (getEvent(index)) {
```

Comment:
The problem here was that if we were in a field-list group and we wanted to get XPath for a question to remember that that question is waiting for an answer it didn't work because we were passing it's index properly like 0,1 for example but then getEvent() (without any parameter) used our current widget to return the XPath which in case of field-list groups is the index of the whole group not of that particular question.

Reply:
Yeah this just seems cleaner as well!